### PR TITLE
fix: Add identity pool ID validation and update Node.js requirement

### PR DESCRIPTION
**Pull request by cgalvan**
## Description

  Fixes two configuration and validation issues:

  1. **Identity Pool ID Validation** - Added validation to ensure identity pool IDs follow the expected `region:guid` format before parsing. Throws clear error messages for invalid
   formats (empty strings, missing colons, malformed structure).

  2. **Node.js Version Update** - Updated minimum Node.js version from 16.x to 20.x, as Node 16 reached EOL in September 2023.

  ## Testing

  - Added 9 new unit tests for identity pool ID validation covering edge cases (empty strings, null/undefined, missing separators, empty parts)
  - All 84 tests pass with 100% coverage for the cognito module
  - Build completes successfully with no TypeScript errors

### CODE REVIEW

**Review by saamzn — APPROVED** [review]

**Review by charsree — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "Eason Huang <yuxuanh@amazon.com>"
   ],
   "engines": {
-    "node": ">= 16.0.0"
+    "node": ">= 20.0.0"
```
Comment (saamzn) [inline package.json:25]: .

**Review by saamzn — APPROVED** [review]
